### ISSUE TITLE
Problem with Amplify.API.get with special characters on Android

### ISSUE THREAD

**Issue by thecillu** [aws-amplify/amplify-flutter#446]
**Bug**
Amplify.API.get call different paths in Android and iOS if the provided path variables contains special chars!

**To Reproduce**
```dart
  Future<User> getUser(String username) async {
    **String path = '/users/$username';**
    print(path);
    RestOptions restOptions = RestOptions(path: path);
    try {
      **RestOperation restOperation = Amplify.API.get(restOptions: restOptions);**
      RestResponse response = await restOperation.response;
      String responseBody = new String.fromCharCodes(response.data);
      var jsonResponse = convert.jsonDecode(responseBody);
      User user;
      if (jsonResponse != null) {
        user = User.fromJson(jsonResponse);
      }
      return user;
    } on ApiException catch (e) {
      print('GET call failed: $e');
    }
  }
```
If the username variables contains special chars, in call 2 different paths in iOS and Android.

Example:
If username equals to 'name.surname+test1@gmail.com', 

- in iOS it call correctly the URL: https://hostname/users/name.surname+test1@gmail.com
- in Android it calls the wrong path: https://hostname/users/name.surname%20test1@gmail.com

**Expected behavior**
Same behaviour in both the platform.

**Platform**
This issue is reproducible in:
[X] Android
[] iOS

**Comment by haverchuck** [comment]
Possibly related to this[ amplify-android ](https://github.com/aws-amplify/amplify-android/issues/668)issue.

**Comment by Jordan-Nelson** [comment]
This has been resolved and should be available in the next major release.

**Comment by Jordan-Nelson** [comment]
Hello - Apologies for the delay on this.

There was a [fix in Amplify-Android](https://github.com/aws-amplify/amplify-android/pull/1252) that I believe will resolve this. We had upgraded the version of Amplify-Android used in Amplify-Flutter to a version that would include this fix, but due to a bug we had downgrade it for the latest release.

The latest stable version of Amplify-Flutter (0.5.0) depend on Amplify-Android 1.33.0. The fix is in version 1.35.1. I am going to remove the pending-release tag until we pull back in 1.35.1 or higher.

**Comment by Jordan-Nelson** [comment]
@thecillu - The fix should have been included in v0.6.2 of Amplify-Flutter. If you are experiencing this issue with a recent version of Amplify-Flutter, please let us know.
